static/app/components/core/button/index.tsx

```diff
@@ -1,102 +1,23 @@
 import isPropValid from '@emotion/is-prop-valid';
-import type {SerializedStyles, Theme} from '@emotion/react';
-import {css} from '@emotion/react';
 import styled from '@emotion/styled';
 
-import {type LinkButtonProps} from 'sentry/components/core/button/linkButton';
-import {Tooltip, type TooltipProps} from 'sentry/components/core/tooltip';
+import {Tooltip} from 'sentry/components/core/tooltip';
 import InteractionStateLayer from 'sentry/components/interactionStateLayer';
-import type {SVGIconProps} from 'sentry/icons/svgIcon';
 import {IconDefaultsProvider} from 'sentry/icons/useIconDefaults';
-import HookStore from 'sentry/stores/hookStore';
 import {space} from 'sentry/styles/space';
 
-import {getChonkButtonStyles} from './index.chonk';
+import {
+  DO_NOT_USE_BUTTON_ICON_SIZES as BUTTON_ICON_SIZES,
+  DO_NOT_USE_getButtonStyles as getButtonStyles,
+} from './styles';
+import {DO_NOT_USE_getChonkButtonStyles as getChonkButtonStyles} from './styles.chonk';
+import type {
+  DO_NOT_USE_ButtonProps as ButtonProps,
+  DO_NOT_USE_CommonButtonProps as CommonButtonProps,
+} from './types';
+import {useButtonFunctionality} from './useButtonFunctionality';
```

Comment:
I would love to come up with a better way of marking exports as "module private". I think we could maybe lint for this, but then we need to come up with some way to avoid the LSP from auto-importing them.

Reply:
separate packages & `exports` in package.json is the way